src/Compiler/Microsoft.CodeAnalysis.Razor.Compiler/src/SourceGenerators/RazorSourceGenerator.cs

```diff
@@ -12,7 +12,9 @@
 
 namespace Microsoft.NET.Sdk.Razor.SourceGenerators
 {
+#pragma warning disable RS1041 // This compiler extension should not be implemented in an assembly with target framework '.NET 8.0'. References to other target frameworks will cause the compiler to behave unpredictably.
```

Comment:
💡 Would be good to add a comment explaining why Razor can do this but if anyone else tries it, their source generator will not work.

Reply:
I frankly don't really know either, I assume @chsienki does.

Reply:
@333fred 🤷 I am also unsure heh, I am surprised we're even building the generator for .NET 8. It's possible it's multitargeted for VS Code hosting but given that we don't use any APIs higher than `netstandard2.0` I don't see why it would make any difference. 

My honest guess is that its just using one of the global MSBuild variables to set target framework and is accidentally picking up the .NET 8 TF.

//cc @maryamariyan and @davidwengier in case they know of a reason.

Reply:
Ohhhh, I see. It's because when @jjonescz did the assembly collapsing we moved the generator into MS.CA.Razor.Compiler as well, rather than being in a separate assembly. That happens to build for multiple TFs, but we should only be using the netstandard2.0 one for the generator.

We should probably check that's how we're actually packaging it.